### ISSUE TITLE
Stitching without Federation

### ISSUE THREAD

**Issue by winsvu** [ghostdogpr/caliban#813]
Hi,

I know Federation is a better way to extend services/schemas, however, I need to stitch a few schemas together without using Federation. Is there any way to do it? do you guys have any examples?

**Comment by paulpdaniels** [comment]
@winsvu It depends on how you mean to stitch them together (and where thee schema's are residing). Currently the only support for stitching built into caliban is to use the `|+|` operator on two `GraphQL`s this will perform a right hand merge of all of the top level fields and will union together the the types from both schema's. But that won't really help you if your schema's are remote since there isn't any support for that at the moment in caliban.

**Comment by winsvu** [comment]
Thanks, @paulpdaniels 

All my schemas are defined in separate projects, then we include them as dependencies in a single project that we call gateway. This gateway is where I need to be able to add new fields and new resolvers because I need data from other schemas/services. They are not running remotely, that is why Federation does not work for me in this case. 

I am migrating from Sangria to Caliban, and in Sangria I define fields using something like this
```scala
      AddFields(
        Field("totalIssues",
          LongType,
          resolve = c => {
            IssueDAO.getTotalIssues(c.value.jourid.get)
          }
        ),
        Field("issues",
          ListType(IssueType),
...
```

That way to define fields in Sangria allows me to extend the Schema later on, and to add resolvers that might call other services. In Caliban once I defined a case class, I am not able to add more fields and resolvers in my gateway project.
I could use Federation, and I like the way it works, however, it requires the **@apollo/gateway** and **BlazeServerBuilder** for every schema, and I don't want to use node and I want to run just one server. @paulpdaniels does it make sense?

I'll give a try to ```|+|```, but not sure that's what I need.

**Comment by paulpdaniels** [comment]
@winsvu Yes makes sense, one possible way I can think of (sort of) emulating the behavior would be to add some methods to "extend" the schema

```scala
  def combine[R1 <: R](other: Schema[R1, T]): Schema[R1, T] = new Schema[R1, T] {

    override protected[this] def toType(isInput: Boolean, isSubscription: Boolean): __Type =
      self.toType_(isInput, isSubscription) |+| other.toType_(isInput, isSubscription)

    override def resolve(value: T): Step[R1] =
      Step.mergeRootSteps(self.resolve(value), other.resolve(value))

  }

  def extend[R1 <: R](fields: FieldAttributes => List[(__Field, T => Step[R1])]): Schema[R1, T] =
    combine(Schema.obj(self.asType.name.getOrElse(""))(fields))
```

This is a bit hacky since I haven't fully thought through this use case, but I _think_ this would allow you to define additional fields on an existing schema by doing:

```scala
val extended = original.extend(implicit attr =>
  List(
    field("name")(_.name)
  )
)
```

**Comment by ghostdogpr** [comment]
Closed by #896
